blockchain/blockindex.go

```diff
@@ -49,6 +93,9 @@ type blockNode struct {
 	nonce      uint32
 	timestamp  int64
 	merkleRoot chainhash.Hash
+
+	// status is a bitfield representing the validation state of the block
```

Comment:
Should we call out this this is protected by the block index mutex?  The rest of the fields are effectively immutable since they aren't changed after the initial creation and thus don't require a mutex.

However, this field is unique in that it is mutable and changed depending on the state.